### PR TITLE
build: Generate gui.files.go on the fly, remove from repo

### PR DESCRIPTION
**Pull request by calmh**
**NOT TO BE MERGED YET**

### Purpose

Removes gui.files.go from the repo, opting instead to rebuild it when necessary. Has implications on how it's possible to build and not.

This is to give something concrete to talk about and try out for https://forum.syncthing.net/t/rework-build-process-to-avoid-repo-bloat/7054

### Testing

It builds on my computer...

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
Good to merge?

File: build.go [hunk 1]
```diff
@@ -429,8 +468,6 @@ func translate() {
 func transifex() {
 	os.Chdir("gui/default/assets/lang")
 	runPrint("go", "run", "../../../../script/transifexdl.go")
```
Comment (lkwg82) [inline build.go]: This relative traversing the tree looks strange to me. What about a convention to run commands from root directory and use something like the environment variable PWD (current working directory).

Relative paths looks very fragile to me (get broken by refactorings etc.). Or is that recommended GO convention?

Comment (canton7) [inline build.go]: I think that convention still stands here? It explicitly chdirs into a subfolder from the root (well, the cwd, which is assumed to be the root), then has a relative path back to the root.

**Comment by calmh** [comment]
Indeed we don't know how to find the build tree root, so I assume it is cwd (i.e. "."). I don't know how to make this more robust? We have other stuff that assumes the build runs from the repo root since before so there is some precedent.

**Comment by calmh** [comment]
@st-review merge this, it's a good thing.

**Comment by st-review** [comment]
OK, merged as 6be4b4999995bc313e1be2444b7d485fba9f903d. Thanks, @calmh!

**Pull request closed without merging**